docs/proposals/20230525-openstack-microversion-support.md

```diff
@@ -0,0 +1,95 @@
+# Configurable/multi microversion support
+
+OpenStack is diverse.
+Multiple versions and configurations exists that can impact what microversions and features that are supported.
+The user (of CAPO) can also to some extent choose what features to make use of.
+The cross section of this is where CAPO is useful:
+
+## Motivation
+
+A CAPO cluster can be configured in many ways and depending on this configuration, certain features may or may not be needed.
+This makes it necessary to know and detect if a feature is used that requires a certain microversion.
+
+It is not enough to check if the user has asked for a specific configuration that would require a certain microversion.
+The requirement can come from the OpenStack instance itself.
+As an example, we have seen environments where all volumes are multi-attachable.
+I.e. it is not possible to create a volume that is not multi-attachable, and thus multi-attach support is a must to be able to do anything in this environment, even if the user doesn't explicitly ask for it.
+
+Currently CAPO has a fixed microversion set.
+It is a take it or leave it situation.
+Either this version is supported by the OpenStack environment or you cannot use CAPO.
+
+This proposal is to make CAPO more flexible in what it supports.
+For example, there could be other versions that doesn't really impact CAPO in any way or there could be cross sections between what the OpenStack environment and CAPO supports, that could be useful to the user if certain features are avoided.
+Here is a simple diagram to try to summarize the situation:
+
+```text
+  ┌────────────────────────────────┐
+  │                                │
+  │                                │
+  │     OpenStack support          │
```

Comment:
Had to look at this a couple of times. Might be better to highlight that this refers to a *specific* OpenStack rather than OpenStack in general, which would presumably encompass everything.

Maybe:
* Features supported by user's OpenStack
* Features supported by CAPO
* Features available to user

Reply:
Updated!